### PR TITLE
Add SSO support for AWS S3 Backend

### LINKED ISSUE #4688: Support AWS SSO for S3 backend
Output of `restic version`
--------------------------

`restic 0.16.2 compiled with go1.21.3 on linux/amd64`

What should restic do differently? Which functionality do you think we should add?
----------------------------------------------------------------------------------
Support AWS SSO logins for authenticating against the S3 backend

What are you trying to do? What problem would this solve?
---------------------------------------------------------
I would like to use restic to backup data on an AWS S3 bucket that is protected using AWS SSO. `AWS_PROFILE` is already supported by restic, but the SSO login doesn't set the env variables directly. Instead, they're written to a json file in `.aws/sso/cache`. I would like restic to read and use them (or find another way to work frictionlessly with SSO).

Did restic help you today? Did it make you happy in any way?
------------------------------------------------------------
Yes, restic does make me happy, so thank you for all your efforts! I feel a lot safer knowing (not just believing) that I have backups available.

### PR DESCRIPTION
**Pull request by konidev20**
What does this PR change? What problem does it solve?
-----------------------------------------------------
- This helps with gh-4688
- The request is to allow a user to use AWS File credentials from the `./aws/sso-cache` to authenticate to AWS S3.

Was the change previously discussed in an issue or on the forum?
----------------------------------------------------------------
Closes #4688

issue from minio/minio-go: https://github.com/minio/minio-go/issues/1930
pr from minio/minio-go: https://github.com/minio/minio-go/issues/1911

Checklist
---------

- [ ] I have read the [contribution guidelines](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#providing-patches).
- [ ] I have [enabled maintainer edits](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added tests for all code changes.
- [ ] I have added documentation for relevant changes (in the manual).
- [ ] There's a new file in `changelog/unreleased/` that describes the changes for our users (see [template](https://github.com/restic/restic/blob/master/changelog/TEMPLATE)).
- [ ] I have run `gofmt` on the code in all commits.
- [ ] All commit messages are formatted in the same style as [the other commits in the repo](https://github.com/restic/restic/blob/master/CONTRIBUTING.md#git-commits).
- [ ] I'm done! This pull request is ready for review.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module github.com/restic/restic
 
+replace github.com/minio/minio-go/v7 => ../minio-go
```
Comment (MichaelEischer) [inline go.mod:3]: That references a local folder on your disk. Why not directly reference the commits from the mentioned minio-go PR?

Comment (konidev20) [reply go.mod:3]: Hey, I was not aware of that method initially. 😅

I will update this to the specific commit hash. 

I have marked the MR in draft for the same.

**Comment by MichaelEischer** [comment]
@konidev20 I'll close that PR for now, feel free to open a new one once there's something to review.

**Comment by konidev20** [comment]
@MichaelEischer sure, I don't think anyone is working on that PR from the MinIO.

**Pull request closed without merging**